src/OverlayTrigger.js

```diff
@@ -0,0 +1,265 @@
+    return {
+      defaultOverlayShown: false,
+      trigger: ['hover', 'focus']
+    };
+  },
+
+  getInitialState() {
+    return {
+      isOverlayShown: this.props.defaultOverlayShown
+    };
+  },
+
+  show() {
+    this.setState({
+      isOverlayShown: true
+    });
+  },
+
+  hide() {
+    this.setState({
+      isOverlayShown: false
+    });
+  },
+
+  toggle() {
+    if (this.state.isOverlayShown) {
+      this.hide();
+    } else {
+      this.show();
+    }
+  },
+
+  componentDidMount(){
+    this._mountNode = document.createElement('div');
+    this.renderOverlay();
+  },
+
+  renderOverlay() {
+    ReactDOM.unstable_renderSubtreeIntoContainer(
+      this, this._overlay, this._mountNode
+    );
+  },
+
+  componentWillUnmount() {
+    ReactDOM.unmountComponentAtNode(this._mountNode);
+    this._mountNode = null;
+    clearTimeout(this._hoverDelay);
+  },
+
+  componentDidUpdate(){
+    if (this._mountNode) {
+      this.renderOverlay();
+    }
+  },
+
+  getOverlayTarget() {
+    return ReactDOM.findDOMNode(this);
+  },
+
+  getOverlay() {
+    let overlayProps = {
+      ...pick(this.props, Object.keys(Overlay.propTypes)),
+      show: this.state.isOverlayShown,
+      onHide: this.hide,
+      target: this.getOverlayTarget,
+      onExit: this.props.onExit,
+      onExiting: this.props.onExiting,
+      onExited: this.props.onExited,
+      onEnter: this.props.onEnter,
+      onEntering: this.props.onEntering,
+      onEntered: this.props.onEntered
+    };
+
+    let overlay = cloneElement(this.props.overlay, {
+      placement: overlayProps.placement,
+      container: overlayProps.container
+    });
+
+    return (
+      <Overlay {...overlayProps}>
+        { overlay }
+      </Overlay>
+    );
+  },
+
+  render() {
+    const trigger = React.Children.only(this.props.children);
+
+    const props = {
+      'aria-describedby': this.props.overlay.props.id
+    };
+
+    // create in render otherwise owner is lost...
+    this._overlay = this.getOverlay();
+
+    props.onClick = createChainedFunction(trigger.props.onClick, this.props.onClick);
+
+    if (isOneOf('click', this.props.trigger)) {
+      props.onClick = createChainedFunction(this.toggle, props.onClick);
+    }
+
+    if (isOneOf('hover', this.props.trigger)) {
+      warning(!(this.props.trigger === 'hover'),
+        'Specifying only the `"hover"` trigger limits the visibilty of the overlay to just mouse users. ' +
+        'Consider also including the `"focus"` trigger so that touch and keyboard only users can see the overlay as well.');
+
+      props.onMouseOver = createChainedFunction(this.handleDelayedShow, this.props.onMouseOver);
+      props.onMouseOut = createChainedFunction(this.handleDelayedHide, this.props.onMouseOut);
+    }
+
+    if (isOneOf('focus', this.props.trigger)) {
+      props.onFocus = createChainedFunction(this.handleDelayedShow, this.props.onFocus);
+      props.onBlur = createChainedFunction(this.handleDelayedHide, this.props.onBlur);
```

Comment:
Not your bug, but see: https://github.com/react-bootstrap/react-bootstrap/issues/1140#issuecomment-133763740

Assuming we get this fixed in R-B first, we'll have to backport the (mostly trivial) fix.